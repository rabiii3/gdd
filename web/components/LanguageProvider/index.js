import { compose } from 'recompose';
import { connect } from 'react-redux';
import component from './component';
import { getLocale } from '../../selectors/language';

const mapStateToProps = state => ({
  locale: getLocale(state),
});

export const enhance = compose(connect(mapStateToProps));

export default enhance(component);
