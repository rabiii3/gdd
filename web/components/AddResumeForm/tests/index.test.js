import React from 'react';
import { Router } from 'react-router-dom';
import { createBrowserHistory } from 'history';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import LanguageProvider from '../../LanguageProvider';
import configureStore from '../../../store/configureStore.test';
import { enhance } from '../';

describe('web |Â components | modalandform | index', () => {
  describe('render', () => {
    it('should render', () => {
      const Component = () => <div />;
      const EnhancedComponent = enhance(Component);
      const state = { language: { locale: 'en' } };
      const store = configureStore(state, null);
      const history = createBrowserHistory();

      const wrapper = mount(
        <Provider store={store}>
          <LanguageProvider messages={{}}>
            <Router history={history}>
              <EnhancedComponent />
            </Router>
          </LanguageProvider>
        </Provider>,
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
});
