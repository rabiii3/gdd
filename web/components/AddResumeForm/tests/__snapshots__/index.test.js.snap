// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`web |Â components | modalandform | index render should render 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(LanguageProvider)
    messages={Object {}}
  >
    <LanguageProvider
      dispatch={[Function]}
      locale="en"
      messages={Object {}}
    >
      <IntlProvider
        key="en"
        locale="en"
      >
        <Router
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
        >
          <WithStyles(Component)>
            <Component
              classes={
                Object {
                  "paper": "Component-paper-1",
                }
              }
            >
              <div />
            </Component>
          </WithStyles(Component)>
        </Router>
      </IntlProvider>
    </LanguageProvider>
  </Connect(LanguageProvider)>
</Provider>
`;
